src/content/advanced/storage/azure-pvc-encryption/index.md

```diff
@@ -0,0 +1,78 @@
+In order to increase the security of your data persisting on PVs you can facilitate [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/) service as an encryption key provider.
+
+## Azure Key Vault setup
+
+Prerequisite for next steps is to have a deployed Key Vault instance either in your clusters Resource Group or in your account. Please follow the [official docs](https://docs.microsoft.com/en-us/azure/key-vault/general/overview) to get started.
+
+## Create first encryption key
+
+After successful deployment we can move to creating encryption key that will be used to encrypt data on your PVs.
+
+Add a new key with key type, RSA key size, activation or expiration date of your choosing in the Key Vault directory.
+
+When creation is finished, the Key Identifier should look similar to:
+```https://name-of-your-key-vault-instance.vault.azure.net/keys/test-vault-enc-key/xxxaaadddwww222111dddaaa23456789```
+
+## Create Disk Encryption Set
+
+Next step is to create a Azure Disk Encryption Set with the key generated in previous step. You can use Azure CLI or navigate in the Portal by adding a new Disk Encryption Set resource.
+
+When creating the Set, you can specify encryption type which should use only Customer-Managed keys or provide a double encryption with additional Platform-Managed keys.
+Regardless of your choice, please specify the key you have just created to be used in the Set.
+
+Remember to add access permissions to the Key Vault instance you have created for the created Disk Encryption Set as it will need it to actually access and read the key after creation.
+
+In the last step you will need the Resource ID of the created Disk Encryption Set looking like:
+```/subscriptions/sub_id/resourceGroups/rg_where_key_vault_is/providers/Microsoft.Compute/diskEncryptionSets/name_of_your_disk_encryption_set```
+
+## Create Storage Class with Encryption
+
+Last step is to create your own Storage Class with the defined encryption set.
+
+After adjusting with your Encryption Set Resource ID, you will be able to apply following code:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1  
+metadata:
+  name: EncryptedDisks
+provisioner: kubernetes.io/azure-disk
+parameters:
+  skuname: Standard_LRS
+  kind: managed
+  diskEncryptionSetID: "/subscriptions/sub_id/resourceGroups/rg_where_key_vault_is/providers/Microsoft.Compute/diskEncryptionSets/name_of_your_disk_encryption_set"
+```
+
+From now on your Persistent Volumes using this storage class will have encrypted data with your Customer-Managed key.
+Final step is to add the Managed Identities (Service Principals) per Virtual Machine Scale Set that are created both for master and for all node pools in your cluster to also have access to the Key Vault instance in order to be able to read the encryption key.
```

Comment:
The previous step said it was the "last step" so maybe another transition is better for either this one or the above step
```suggestion
The final step is to add the Managed Identities (Service Principals) per Virtual Machine Scale Set that are created both for master and for all node pools in your cluster to also have access to the Key Vault instance in order to be able to read the encryption key.
```